### PR TITLE
all: ability to connect to ES v20.6.0+

### PR DESCRIPTION
**Pull request by ahjohannessen**
- add gossip terminology mapping in `NodeState` in order to support
   both v5.x and v20.x servers.

 - add node states `ReadOnlyLeaderless`, `PreReadOnlyReplica` and
   `ReadOnlyReplica` that are in ES v20.x - As the JVM client does
   not have a concept of node preference as the .NET client, these
   states have lower priority than `Follower` and `Leader`.

 - remove `isReplica` as it is not used anywhere.

 - add `IsReadOnly` to `NotHandled` protobuf definition in order to
   have `SimpleOperation` handle `NotHandled(IsReadOnly)` failures.

 - change ci to test against ES20.6.0 and ES5.0.8

 - add `isES20Series` condition to `TransactionITest` because of
   EventStore/EventStore@0db40ab#diff-b20ab1e660cda65b9fb97a8271bfc669R77
   This makes the `idempotency` test pass on 5.0.x and 20.6.x nodes.

 - fix various compiler warnings.

### CODE REVIEW

**Comment by ahjohannessen** [comment]
ES 20.6.0 integration fails transaction test due to this https://github.com/EventStore/EventStore/commit/0db40ab6a1b667060fef9948fc5135b90d4a9b34#diff-b20ab1e660cda65b9fb97a8271bfc669

**Comment by ahjohannessen** [comment]
@t3hnar WDYT?

**Comment by ahjohannessen** [comment]
Do not merge yet, ran a cluster with es20.6.0 nodes and learned that the json is not the same wrt. external / internal http and needs some sort of mapping or similar. Not a biggie, will look into it :)

**Comment by ahjohannessen** [comment]
Should be good now. Did the necessary compatibility stuff to handle pre and post ES20 wrt. json stuff. Also added cluster integration tests to travis. Travis should pass, tests pass with clusters with ES5.x and ES20.6.x locally.

File: .travis.yml [hunk 1]
```diff
@@ -21,16 +21,53 @@ after_success:
...
 jobs:
   include:
-    - stage: integration
-      scala: 2.13.1
+
+    - stage: integration-20.x
+      scala: 2.13.3
+      env: AKKA_TEST_TIMEFACTOR=1.5 AKKA_TEST_LOGLEVEL=OFF ES_TEST_ADDRESS_PORT=1114 ES_TEST_HTTP_PORT=2114 ES_TEST_IS_20_SERIES=true
+      before_install:
+        - docker pull quay.io/ahjohannessen/eventstore-20.6.0-bionic
```
Comment (t3hnar) [inline .travis.yml:29]: @ahjohannessen are we missing an official image?

**Comment by t3hnar** [comment]
@ahjohannessen  thanks, it looks good

File: .travis.yml (hunk 1 above)
Comment (ahjohannessen) [reply .travis.yml:29]: @t3hnar Just for a while, when they sync with docker hub I’ll change to that :)

**Pull request merged**